src/modules/ctc/src/ctc.cc

```diff
@@ -64,6 +64,53 @@ types::Error Ctc::UpdateSuggestedSpeedAndAuthority(const types::TrainId train_id
     return error;
 }
 
+types::Error Ctc::SetBlockStates(const types::TrackId track, const std::vector<types::BlockState> &block_states)
+{
+    types::Error   error     = types::Error::ERROR_INVALID_FORMAT;
```

Comment:
Return error codes based on the errors encountered.  For example, in the case of an invalid block ID, this method will return the error for invalid format rather than the error for invalid block ID.

Reply:
resolved